flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/writer/BigQueryDefaultWriter.java

```diff
@@ -105,6 +141,9 @@ void validateAppendResponse(AppendInfo appendInfo) {
                 logAndThrowFatalException(response.getError().getMessage());
             }
             totalRecordsWritten += recordsAppended;
+            // the request succeeded without errors (records are in BQ)
+            this.numberOfRecordsWrittenToBigQuery.inc(recordsAppended);
+            this.numberOfRecordsWrittenToBigQuerySinceCheckpoint.inc(recordsAppended);
```

Comment:
This is not true. `sendAppendRequest` has literally juts sent the request to append data, but we're not sure if this succeeded until validation of the response is performed later. That validation is defined in this file. You should increment the write count at that stage, not here

Reply:
Counters are incremented under `validateAppendResponse()` only. Once the response is checked for errors.

Reply:
Ah got it. My bad. Thanks for clarifying.